File: lib/data/dto/stats_dto.dart
class DailyStatsDto {
  final int screenTimeMinutes;
  final int pointsEarned;
  final int pointsCollected;
  final int uncollectedPoints;
  final List<dynamic> appBreakdown;
  final DateTime date;

  const DailyStatsDto({
    required this.screenTimeMinutes,
    required this.pointsEarned,
    required this.pointsCollected,
    required this.uncollectedPoints,
    required this.appBreakdown,
    required this.date,
  });

  factory DailyStatsDto.fromJson(Map<String, dynamic> json) => DailyStatsDto(
        screenTimeMinutes: (json['screenTimeMinutes'] as num?)?.toInt() ?? 0,
        pointsEarned: (json['pointsEarned'] as num?)?.toInt() ?? 0,
        pointsCollected: (json['pointsCollected'] as num?)?.toInt() ?? 0,
        uncollectedPoints: (json['uncollectedPoints'] as num?)?.toInt() ?? 0,
        appBreakdown: (json['appBreakdown'] as List<dynamic>?) ?? [],
        date: json['date'] != null
            ? DateTime.tryParse(json['date'].toString()) ?? DateTime.now()
            : DateTime.now(),
      );
}

class WeeklyStatsEntryDto {
  final String dayLabel;
  final int minutes;
  final int points;
  final bool isToday;

  const WeeklyStatsEntryDto({
    required this.dayLabel,
    required this.minutes,
    required this.points,
    required this.isToday,
  });

  factory WeeklyStatsEntryDto.fromJson(Map<String, dynamic> json) =>
      WeeklyStatsEntryDto(
        dayLabel: json['dayLabel'] as String,
        minutes: (json['minutes'] as num).toInt(),
        points: (json['points'] as num).toInt(),
        isToday: json['isToday'] as bool,
      );
}

class StreakDto {
  final int currentStreak;
  final int longestStreak;
  final double multiplier;
  final bool collectedToday;
  final String nextMilestoneLabel;

  const StreakDto({
    required this.currentStreak,
    required this.longestStreak,
    required this.multiplier,
    required this.collectedToday,
    required this.nextMilestoneLabel,
  });

  factory StreakDto.fromJson(Map<String, dynamic> json) => StreakDto(
        currentStreak: (json['currentStreak'] as num?)?.toInt() ?? 0,
        longestStreak: (json['longestStreak'] as num?)?.toInt() ?? 0,
        multiplier: (json['multiplier'] as num?)?.toDouble() ?? 1.0,
        collectedToday: json['collectedToday'] as bool? ?? false,
        nextMilestoneLabel:
            (json['nextMilestoneLabel'] ?? 'Keep going!').toString(),
      );
}

File: backend/src/stats/dto/stats.dto.ts
import { ApiProperty } from '@nestjs/swagger';

export class DailyStatsDto {
  @ApiProperty({ description: 'Total social media minutes tracked today' })
  screenTimeMinutes: number;

  @ApiProperty({ description: 'Points earned from screen time today' })
  pointsEarned: number;

  @ApiProperty({ description: 'Points collected (moved to wallet) today' })
  pointsCollected: number;

  @ApiProperty({ description: 'Current uncollected points (capped)' })
  uncollectedPoints: number;

  @ApiProperty({ description: 'Per-app breakdown of screen time' })
  appBreakdown: AppScreenTimeDto[];

  @ApiProperty()
  date: Date;
}

export class AppScreenTimeDto {
  @ApiProperty()
  appId: string;

  @ApiProperty()
  appName: string;

  @ApiProperty({ description: 'Minutes spent today' })
  minutes: number;
}

export class WeeklyStatsEntryDto {
  @ApiProperty()
  dayLabel: string;

  @ApiProperty({ description: 'Screen time minutes for this day' })
  minutes: number;

  @ApiProperty({ description: 'Points earned this day' })
  points: number;

  @ApiProperty()
  isToday: boolean;
}

export class ReferralStatsOverviewDto {
  @ApiProperty({ description: 'Number of successful direct invites (children)' })
  directInvites: number;

  @ApiProperty({ description: 'Number of grandchild invites (invitees of your invitees)' })
  grandChildInvites: number;

  @ApiProperty({ description: 'Total invites across both levels' })
  totalInvites: number;
}

export class StreakDto {
  @ApiProperty({ description: 'Consecutive days the user has collected points' })
  currentStreak: number;

  @ApiProperty()
  longestStreak: number;

  @ApiProperty({ description: 'Bonus multiplier based on collection streak' })
  multiplier: number;

  @ApiProperty({ description: 'Whether user has collected points today' })
  collectedToday: boolean;

  @ApiProperty()
  nextMilestoneLabel: string;
}
